test/functional/wallet_importdescriptors.py

```diff
@@ -666,5 +666,28 @@ def run_test(self):
                               success=True,
                               warnings=["Unknown output type, cannot set descriptor to active."])
 
+        self.log.info("Test importing a descriptor to an encrypted wallet")
+
+        descriptor = {"desc": descsum_create("pkh(" + xpriv + "/1h/*h)"),
+                              "timestamp": "now",
+                              "active": True,
+                              "range": [0,4000],
+                              "next_index": 4000}
+
+        self.nodes[0].createwallet("temp_wallet", blank=True, descriptors=True)
+        temp_wallet = self.nodes[0].get_wallet_rpc("temp_wallet")
+        temp_wallet.importdescriptors([descriptor])
+        self.generatetoaddress(self.nodes[0], COINBASE_MATURITY + 1, temp_wallet.getnewaddress())
+        self.generatetoaddress(self.nodes[0], COINBASE_MATURITY + 1, temp_wallet.getnewaddress())
+
+        self.nodes[0].createwallet("encrypted_wallet", blank=True, descriptors=True, passphrase="passphrase")
+        encrypted_wallet = self.nodes[0].get_wallet_rpc("encrypted_wallet")
+        encrypted_wallet.walletpassphrase("passphrase", 1)
+
+        descriptor["timestamp"] = 0
+        descriptor["next_index"] = 0
+        encrypted_wallet.importdescriptors([descriptor])
```

Comment:
In fcaaa5fd905a21298bd69343c9756e32a70870e6 "test: test rescanning encrypted wallets"

I suspect that this test is going to be the source of many intermittent failures in the future. We already observe a failure in CI for this test. Since this is testing a race condition, the speed of the machine executing the test becomes a significant factor, and having sanitizers being run can cause the test to fail.

I think we might be able to workaround this by using a batched RPC request as IIRC those are run in a single thread sequentially.

Reply:
I agree that this test would have been problematic. I have changed the `importdescriptors` and `rescanblockchain` tests added in this PR to use a batched RPC request.